apps/e2e/ag-grid-storybook/src/app/data-entry-grid/data-entry-grid.component.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import {
+  SkyTheme,
+  SkyThemeMode,
+  SkyThemeService,
+  SkyThemeSettingsChange,
+} from '@skyux/theme';
+
+import { GridReadyEvent } from 'ag-grid-community';
+import { BehaviorSubject } from 'rxjs';
+
+import { DataEntryGridComponent } from './data-entry-grid.component';
+import { DataEntryGridModule } from './data-entry-grid.module';
+
+describe('DataEntryGridComponent', () => {
+  let component: DataEntryGridComponent;
+  let fixture: ComponentFixture<DataEntryGridComponent>;
+  let mockThemeSvc: {
+    settingsChange: BehaviorSubject<SkyThemeSettingsChange>;
+  };
+
+  beforeEach(() => {
+    mockThemeSvc = {
+      settingsChange: new BehaviorSubject<SkyThemeSettingsChange>({
+        currentSettings: {
+          theme: SkyTheme.presets.default,
+          mode: SkyThemeMode.presets.light,
+        },
+        previousSettings: undefined,
+      }),
+    };
+    TestBed.configureTestingModule({
+      imports: [DataEntryGridModule],
+      providers: [
+        {
+          provide: SkyThemeService,
+          useValue: mockThemeSvc,
+        },
+      ],
+    });
+    fixture = TestBed.createComponent(DataEntryGridComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', async () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should create edit-lookup variation', async () => {
+    component.variation = 'edit-lookup';
+    component.ngOnInit();
+    expect(component).toBeTruthy();
+  });
+
+  it('should become ready', async () => {
+    Object.values(component.gridOptions).forEach((gridOptions) => {
+      gridOptions.onGridReady({} as GridReadyEvent);
+    });
+    expect(component.active$.value).toBe(true);
+    component.ngAfterViewInit();
+    fixture.detectChanges();
+    await fixture.whenStable();
+    expect(component).toBeTruthy();
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    expect(component.ready.value).toBe(true);
```

Comment:
The `ngAfterViewInit` calls `delay(1000)` before updating `ready` because AG Grid needs extra time to complete its rendering.